src/internal/lastValueFrom.ts

```diff
@@ -40,24 +48,28 @@ import { EmptyError } from './util/EmptyError';
  * ```
  *
  * @param source the observable to convert to a promise
+ * @param config a configuration object to define the `defaultValue` to use if the source completes without emitting a value
  */
-export function lastValueFrom<T>(source: Observable<T>) {
-  return new Promise<T>((resolve, reject) => {
+export function lastValueFrom<T, D>(source: Observable<T>, config?: LastValueFromConfig<D>) {
+  const hasConfig = arguments.length >= 2;
```

Comment:
For non-TS users, (or bad TS users) this can cause runtime errors: `lastValueFrom(promise, undefined)`

It doesn't really `hasConfig` unless `typeof config === 'object'`, I guess.

Reply:
More realistically, I can see people doing things like:

```
const maybeConfig = externalConfig?.defaultValueConfig // maybe undefined

lastValueFrom(promise, maybeConfig);
```

It would be annoying to need to do this:

```
if (maybeConfig) {
  lastValueFrom(promise, maybeConfig);
} else {
  lastValueFrom(promise);
}
```

Reply:
Nice catch. Done.